File: _vim/template/ruby/lib/template.rb
<%=
namespaces = File.dirname('FILE_PATH'.split('lib/', 2)[1])
  .gsub(/(_|\b)([a-z0-9])/) { $2.upcase }
  .split('/')
  .reject { |n| n == '' }

class_name = File.basename('FILE_NAME', '.rb')
  .gsub(/(_|\b)([a-z0-9])/) { $2.upcase }
  .delete('_')

out = <<EOS.chomp
class #{class_name}

  <+CURSOR+>

end
EOS

namespaces.reverse.each do |namespace|
  out = <<EOS.chomp
module #{namespace}
#{out.gsub(/^/, '  ')}
end
EOS
end

out.gsub(/^[ ]+$/, '')
%>

File: nvim/template/go/api/template.go
<%=
class_name = File.basename('FILE_NAME', '.go')
  .gsub(/(_|\b)([a-z0-9])/) { $2.upcase }
  .delete('_')

out = <<EOS.chomp
package api

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type #{class_name} struct{}

func (#{class_name}) <+CURSOR+>(c *gin.Context) {

}
EOS

out
%>
